ray-operator/controllers/raycluster_controller.go

```diff
@@ -242,7 +311,56 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 				return err
 			}
 		}
+	} else {
+		// head node is exactly 1, but we need to check if it has been changed
+		res := r.needRemoveOldPod(instance.Spec.HeadGroupSpec.Template, headPods.Items[0])
+		if res {
+			log.Info(fmt.Sprintf("need to delete old head pod %s", headPods.Items[0].Name))
+			if err := r.Delete(context.TODO(), &headPods.Items[0]); err != nil {
+				return err
+			}
+			// delete all worker pods
```

Comment:
Do we want to delete workers if they are not changed? Once new head is back, I think worker will rejoin the new head.

Reply:
This is a good point. I was also thinking about the best way to do this. But anyway, I think lets just remove this extra logic for now.